homeassistant/components/solarlog/const.py

```diff
@@ -17,83 +30,190 @@
+        key="yield_total",
+        json_key="yieldTOTAL",
+        name="yield total",
+        icon="mdi:solar-power",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        state_class=STATE_CLASS_TOTAL_INCREASING,
+    ),
+    SolarLogSensorEntityDescription(
+        key="consumption_ac",
+        json_key="consumptionAC",
+        name="consumption AC",
+        native_unit_of_measurement=POWER_WATT,
+        device_class=DEVICE_CLASS_POWER,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    SolarLogSensorEntityDescription(
+        key="consumption_day",
+        json_key="consumptionDAY",
+        name="consumption day",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        device_class=DEVICE_CLASS_ENERGY,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    SolarLogSensorEntityDescription(
+        key="consumption_yesterday",
+        json_key="consumptionYESTERDAY",
+        name="consumption yesterday",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        device_class=DEVICE_CLASS_ENERGY,
+    ),
+    SolarLogSensorEntityDescription(
+        key="consumption_month",
+        json_key="consumptionMONTH",
+        name="consumption month",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        device_class=DEVICE_CLASS_ENERGY,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    SolarLogSensorEntityDescription(
+        key="consumption_year",
+        json_key="consumptionYEAR",
+        name="consumption year",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        device_class=DEVICE_CLASS_ENERGY,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    SolarLogSensorEntityDescription(
+        key="consumption_total",
+        json_key="consumptionTOTAL",
+        name="consumption total",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        device_class=DEVICE_CLASS_ENERGY,
+        state_class=STATE_CLASS_TOTAL_INCREASING,
+    ),
+    SolarLogSensorEntityDescription(
+        key="total_power",
+        json_key="totalPOWER",
+        name="total power",
+        icon="mdi:power-plug",
+        native_unit_of_measurement="Wp",
+    ),
+    SolarLogSensorEntityDescription(
+        key="alternator_loss",
+        json_key="alternatorLOSS",
+        name="alternator loss",
+        icon="mdi:solar-power",
+        native_unit_of_measurement=POWER_WATT,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    SolarLogSensorEntityDescription(
+        key="capacity",
+        json_key="CAPACITY",
+        name="capacity",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=DEVICE_CLASS_POWER_FACTOR,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    SolarLogSensorEntityDescription(
+        key="efficiency",
+        json_key="EFFICIENCY",
+        name="efficiency",
+        icon="mdi:solar-power",
+        native_unit_of_measurement=f"% {POWER_WATT}/{POWER_WATT}p",
```

Comment:
Isn't the unit just `%`?

Reply:
No, according to the solarlog API, it is W/Wp

Reply:
Fixed this. Efficiency is AC power / DC power * 100 %, so W/W * 100%. So, the unit is PERCENTAGE, you are right. I also added the device class "power_factor". 

The "capacity" sensor is according to the sunwatcher Pypi package defined as DC power (in Watt) / total power of the system (in Wp (Watt-peak)). So I changed the unit of measurement to "W/Wp".